scilla/src/backend_collector.rs

```diff
@@ -0,0 +1,365 @@
+use std::collections::HashMap;
+
+use evm::backend::Backend;
+use evm_ds::protos::evm_proto::{Apply, EvmResult, Storage};
+use primitive_types::{H160, H256, U256};
+use serde_json::Value;
+use sha3::{Digest, Keccak256};
+use tracing::*;
+
+pub type Address = H160;
+type AccountStorage = HashMap<H256, H256>;
```

Comment:
I would argue (but certainly up for debate), that you can essentially avoid doing this by using the mapping algorithm here (previously called `compression`) and essentially point it at storage to provide an abstraction where the caller sees it as `(String, String)`. You know whether you need to do this because the `Account` flags whether it is scilla or not.

WRT key value pairs, the tests I am running require them, and so there is a scheme to recover the original keys, too. it is called in `node.rs` as `get_scilla_kv_pairs` (not sure if that's the best name for it).

Reply:
I don't know what your solution is for the z1 storage, but you should also be able to map the z1 storage to this storage using this scheme and everything should work as before.